### PR TITLE
Ports atmos hide rework and plumbing mapping

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/49644
- https://github.com/tgstation/tgstation/pull/50015/

## About The Pull Request
This ports a few features from /tg/. Namely, it makes the hiding part of atmos pipes and machines (that thing where pipes will go under or over tiles) into an actual element and adds a var called "hide". This variable is set to true/false and determines whether it is beneath or above the tile, independent of layering or if the pipe has /hidden/ in it. This allows mappers to set the variable for basically any pipe they want, coloring and layer need not apply.

It also has the side effect of allowing those with VV perms to change whether machines or pipes appear or not, aswell, as the var can be edited ingame. I dont see much of a use for it, but it's easier now.

Plumbing/Ducts can also be properly mapped without doing dumb shit now, also.

Additional code changes should be generally self-explantory, either replacing old code (like update_icon), editing some weird comment spacings, or eliminating runtimes that result from moving HIDE or _plumbing

tl;dr
- You can now map plumbing
- converted cringe hide(T.intact) and other shitty invisibilty, add icon or (intact) calls into AddElement(/datum/element/undertile, TRAIT_T_RAY_VISIBLE)
- code cleanup on aisle 992

## Why It's Good For The Game
Im an amateur mapper personally and the way pipes are now is way too complicated. Mappers should be able to determine whether the pipes appear or not with a single variable.

Also, the side benefit of cutting out a lot of code in favor of an element. Plus this is a requisite for more /tg/ ports.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

epic1
epic2
Proof this PR didnt fuck up any existing layering or hidden pipes.

https://user-images.githubusercontent.com/62388554/210124652-a70c4c35-21a7-4e63-b740-f91aa153c65f.mp4

Ghub didnt like my >10MB video, so heres a fried version. 
As can be seen, the variable has been correctly inherited by pipes and machines that were hidden or unhidden before, as is NULL for items that do not have a set 'hide' var value

</details>

## Changelog
:cl: Time-Green, RKz
add: Plumbing can now (technically) be mapped.
tweak: atmos pipe/machine tile hiding is now a variable, mappers and admins rejoice.
fix: Plumbing should have been mapped long ago.
refactor: quite a few .dm outside atmos pipes, such as satchel bags, have been simplified now that the feature is tacked onto a variable
code: a few update_appearance replacing update_icon
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
Apparentely some old RandomZLevel had shit hardcoded. 

See ya

**Comment by Tsar-Salat** [comment]
bye bye map errors 😎

**Review by PestoVerde322 — DISMISSED** [review]
Mapside looks fine, approved on there

**Review by PowerfulBacon — DISMISSED** [review]

**Comment by PowerfulBacon** [comment]
We should probably testmerge this, but its not required

**Comment by Tsar-Salat** [comment]
> We should probably testmerge this, but its not required

Really my only concern was that pipes ingame wouldn't inherit correctly after I removed their layer definitions but it seemed to work out fine, as seen in the picture.

I guess the only thing to do during the testmerge is seeing if the pipes that I removed layer defines for (a few toxins devices) appear fine in the maps. In my testing they were fine but who knows

**Comment by Tsar-Salat** [comment]
bump

File: code/datums/components/plumbing/_plumbing.dm [hunk 1]
```diff
@@ -27,12 +28,13 @@
...
+	RegisterSignal(parent, list(COMSIG_ATOM_UPDATE_OVERLAYS), .proc/create_overlays) //create overlays also gets called after init (no idea by what it just happens)
 
 	if(start)
-		enable()
-
-	if(use_overlays)
-		create_overlays()
+		//timer 0 so it can finish returning initialize, after which we're added to the parent.
+		//Only then can we tell the duct next to us they can connect, because only then is the component really added. this was a fun one
+		addtimer(CALLBACK(src, .proc/enable), 0)
```
Comment (itsmeow) [inline code/datums/components/plumbing/_plumbing.dm:37]: can't you use LateInitialize hints for this? or are those not available on components?

File: code/datums/components/plumbing/_plumbing.dm [hunk 2]
```diff
@@ -208,12 +229,19 @@
 		net.add_plumber(src, dir)
 		net.add_plumber(P, opposite_dir)
 
+/datum/component/plumbing/proc/hide(atom/movable/AM, intact)
+
```
Comment (itsmeow) [inline code/datums/components/plumbing/_plumbing.dm:233]: weird spacing here. also why is the var named "intact"?

File: code/modules/atmospherics/machinery/pipes/multiz.dm [hunk 3]
```diff
@@ -19,9 +19,6 @@
 	var/mutable_appearance/pipe = null
 	var/obj/machinery/atmospherics/front_node = null
 
-// Do not hide when under floor
-/obj/machinery/atmospherics/pipe/multiz/hide(i)
-	return
```
Comment (itsmeow) [inline code/modules/atmospherics/machinery/pipes/multiz.dm:24]: do these hide now

File: code/datums/components/plumbing/_plumbing.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/datums/components/plumbing/_plumbing.dm:233]: Intact was always the name of the var. The hide element simply inherited the name.

I would rather keep it for compatibility reasons.

File: code/datums/components/plumbing/_plumbing.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/datums/components/plumbing/_plumbing.dm:37]: It wouldn't work here

File: code/modules/atmospherics/machinery/pipes/multiz.dm (hunk 3 above)
Comment (Tsar-Salat) [reply code/modules/atmospherics/machinery/pipes/multiz.dm:24]: Apologies, oversight. I have added "hide = FALSE" var to address
image

**Comment by itsmeow** [comment]
accidentally forgot to TM, huh

**Review by itsmeow — APPROVED** [review]
fuck it we ball

**Comment by itsmeow** [comment]
if this breaks im exploding you, YOLO

**Comment by Rukofamicom** [comment]
> if this breaks im exploding you, YOLO

Spoiler: It broke and has been waiting three weeks for a fix since initially being reported as breaking things. 
Author has declared they have no intention of trying to fix at this time, so I'm opening a revert until they do.

**Comment by itsmeow** [comment]
What is broken?

**Comment by itsmeow** [comment]
Plumbing inputs? Just disable hiding on the pipes placed by the RPD

**Comment by Rukofamicom** [comment]
Plumbing inputs and outputs.

**Comment by Tsar-Salat** [comment]
I commented out the offending proc in #8865.

**Comment by Rukofamicom** [comment]
#8866 properly fixes the issue

~~revert wasn't possible~~
